src/feat/online-feature.h

```diff
@@ -41,6 +41,32 @@ namespace kaldi {
 /// @{
 
 
+/// This class serves as a storage for feature vectors with an option to limit
+/// the memory usage by removing old elements. The deleted frames indices are
+/// "remembered" so that regardless of the MAX_ITEMS setting, the user always
+/// provides the indices as if no deletion was being performed.
+/// This is useful when processing very long recordings which would otherwise
+/// cause the memory to eventually blow up when the features are not being removed.
+class RecyclingVector {
+public:
+  /// By default it does not remove any elements.
+  RecyclingVector(int items_to_hold = -1);
+
+  Vector<BaseFloat> *Retrieve(int index) const;
+
+  void Store(Vector<BaseFloat> *item);
```

Comment:
Please document it very clearly that the `item` becomes owned by this collection and must not be deleted.

Reply:
+1